pkg/tests/apis/helper.go

```diff
@@ -76,6 +78,82 @@ func NewK8sTestHelper(t *testing.T) *K8sTestHelper {
 	return c
 }
 
+type ResourceClientArgs struct {
+	User      User
+	Namespace string
+	GVR       schema.GroupVersionResource
+}
+
+type K8sResourceClient struct {
+	t        *testing.T
+	Args     ResourceClientArgs
+	Resource dynamic.ResourceInterface
+}
+
+// This will set the expected Group/Version/Resource and return the discovery info if found
+func (c *K8sTestHelper) GetResourceClient(args ResourceClientArgs) *K8sResourceClient {
+	c.t.Helper()
+
+	if args.Namespace == "" {
+		args.Namespace = c.namespacer(args.User.Identity.GetOrgID())
+	}
+
+	addr := c.env.Server.HTTPServer.Listener.Addr()
+	baseUrl := fmt.Sprintf("http://%s", addr)
+	login := args.User.Identity.GetLogin()
+	if login != "" && args.User.password != "" {
+		baseUrl = fmt.Sprintf("http://%s:%s@%s", login, args.User.password, addr)
+	}
+
+	config := &rest.Config{
+		Host: baseUrl,
```

Comment:
@toddtreece -- looks like having the basic auth password in the URL is sufficient

Reply:
it looks like `Username` and `Password` are [rest.Config](https://pkg.go.dev/k8s.io/client-go@v0.28.3/rest#Config) options